packages/miniflare/src/plugins/assets/schema.ts

```diff
@@ -14,7 +14,13 @@ export const AssetsOptionsSchema = z.object({
 			directory: PathSchema,
 			binding: z.string().optional(),
 			routerConfig: RouterConfigSchema.optional(),
-			assetConfig: AssetConfigSchema.optional(),
+			assetConfig: AssetConfigSchema.omit({
+				compatibility_date: true,
+				compatibility_flags: true,
+			}).optional(),
 		})
 		.optional(),
+
+	compatibilityDate: z.string().optional(),
```

Comment:
We don't support diverging the asset compat options from the worker's. So might as well pick up the compat options from there rather than asking clients to drill them through the assets config object.